### DISCUSSION TITLE
Welcome to python-hdwallet Discussions!

### DISCUSSION THREAD

**Discussion by meherett** [hdwallet-io/python-hdwallet#6] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by K1du** [comment]
Hi,im eagerly waiting for you to finnish the automation project you started,i would love to get it running 24/7 from a random mnemonic list,also could you setup some aliases to not run it from python3 directly?
I'm learning to use it properly,you have TG?

I apreciate the time and effort that you are putting into this project,Good work and good job.

**Reply by meherett** [reply]
Thanks :) and yes I can prepare, you can contact me via TG send me PM at @meherett username.

**Comment by aliborhani59** [comment]
Hi
I try to install hd-wallet by "pip install hdwallet" but I receive this error:
Collecting hdwallet
  Using cached hdwallet-1.3.2-py3-none-any.whl (44 kB)
Collecting ecdsa<1,>=0.13
  Using cached ecdsa-0.17.0-py2.py3-none-any.whl (119 kB)
Collecting mnemonic<1,>=0.19
  Using cached mnemonic-0.20-py3-none-any.whl (62 kB)
Collecting base58<3,>=2.0.1
  Using cached base58-2.1.0-py3-none-any.whl (5.6 kB)
Collecting pysha3<2,>=1.0.2
  Using cached pysha3-1.0.2.tar.gz (829 kB)
Requirement already satisfied: six>=1.9.0 in c:\spyder\pkgs (from ecdsa<1,>=0.13->hdwallet) (1.15.0)
Building wheels for collected packages: pysha3
  Building wheel for pysha3 (setup.py): started
  Building wheel for pysha3 (setup.py): finished with status 'error'
  Running setup.py clean for pysha3
Note: you may need to restart the kernel to use updated packages.Failed to build pysha3
Installing collected packages: pysha3, mnemonic, ecdsa, base58, hdwallet
    Running setup.py install for pysha3: started
    Running setup.py install for pysha3: finished with status 'error'

  ERROR: Command errored out with exit status 1:
   command: 'C:\spyder\Python\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Ali-772\\AppData\\Local\\Temp\\pip-install-ew4nd0a9\\pysha3_820db6f70bd74df08c2998b57def4e9c\\setup.py'"'"'; __file__='"'"'C:\\Users\\Ali-772\\AppData\\Local\\Temp\\pip-install-ew4nd0a9\\pysha3_820db6f70bd74df08c2998b57def4e9c\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d 'C:\Users\Ali-772\AppData\Local\Temp\pip-wheel-fqpclywv'
       cwd: C:\Users\Ali-772\AppData\Local\Temp\pip-install-ew4nd0a9\pysha3_820db6f70bd74df08c2998b57def4e9c\
  Complete output (9 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build\lib.win-amd64-3.7
  copying sha3.py -> build\lib.win-amd64-3.7
  running build_ext
  building '_pysha3' extension
  error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
  ----------------------------------------
  ERROR: Failed building wheel for pysha3
    ERROR: Command errored out with exit status 1:
     command: 'C:\spyder\Python\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Ali-772\\AppData\\Local\\Temp\\pip-install-ew4nd0a9\\pysha3_820db6f70bd74df08c2998b57def4e9c\\setup.py'"'"'; __file__='"'"'C:\\Users\\Ali-772\\AppData\\Local\\Temp\\pip-install-ew4nd0a9\\pysha3_820db6f70bd74df08c2998b57def4e9c\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\Ali-772\AppData\Local\Temp\pip-record-onz71jqb\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\spyder\Python\Include\pysha3'
         cwd: C:\Users\Ali-772\AppData\Local\Temp\pip-install-ew4nd0a9\pysha3_820db6f70bd74df08c2998b57def4e9c\
    Complete output (9 lines):
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.7
    copying sha3.py -> build\lib.win-amd64-3.7
    running build_ext
    building '_pysha3' extension
    error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'C:\spyder\Python\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Ali-772\\AppData\\Local\\Temp\\pip-install-ew4nd0a9\\pysha3_820db6f70bd74df08c2998b57def4e9c\\setup.py'"'"'; __file__='"'"'C:\\Users\\Ali-772\\AppData\\Local\\Temp\\pip-install-ew4nd0a9\\pysha3_820db6f70bd74df08c2998b57def4e9c\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\Ali-772\AppData\Local\Temp\pip-record-onz71jqb\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\spyder\Python\Include\pysha3' Check the logs for full command output.

can anyone help me pass though it?

**Comment by meherett** [comment]
First, Install Microsoft C++ Build Tools on your windows machine. Pysha3 library requires "Microsoft C++ Build Tools" for the C++ compiler. Thanks!

**Reply by aliborhani59** [reply]
dear meherett tnx for reply,
after install VS 2019 i've got this error:
notice that I use spyder to write my code.
regards

Collecting hdwallet
  Using cached hdwallet-1.3.2-py3-none-any.whl (44 kB)
Collecting base58<3,>=2.0.1
  Using cached base58-2.1.0-py3-none-any.whl (5.6 kB)
Collecting mnemonic<1,>=0.19
  Using cached mnemonic-0.20-py3-none-any.whl (62 kB)
Collecting ecdsa<1,>=0.13
  Using cached ecdsa-0.17.0-py2.py3-none-any.whl (119 kB)
Collecting pysha3<2,>=1.0.2
  Using cached pysha3-1.0.2.tar.gz (829 kB)
Requirement already satisfied: six>=1.9.0 in c:\program files\spyder\pkgs (from ecdsa<1,>=0.13->hdwallet) (1.16.0)
Building wheels for collected packages: pysha3
  Building wheel for pysha3 (setup.py): started
  Building wheel for pysha3 (setup.py): finished with status 'error'
  Running setup.py clean for pysha3
Failed to build pysha3
Installing collected packages: pysha3, mnemonic, ecdsa, base58, hdwallet
    Running setup.py install for pysha3: started
    Running setup.py install for pysha3: finished with status 'error'
Note: you may need to restart the kernel to use updated packages.
  ERROR: Command errored out with exit status 1:
   command: 'C:\Program Files\Spyder\Python\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Ali-772\\AppData\\Local\\Temp\\pip-install-1kuk9w3o\\pysha3_ddd48db2c66944edaf8e695c3fb85cad\\setup.py'"'"'; __file__='"'"'C:\\Users\\Ali-772\\AppData\\Local\\Temp\\pip-install-1kuk9w3o\\pysha3_ddd48db2c66944edaf8e695c3fb85cad\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d 'C:\Users\Ali-772\AppData\Local\Temp\pip-wheel-agubmz6u'
       cwd: C:\Users\Ali-772\AppData\Local\Temp\pip-install-1kuk9w3o\pysha3_ddd48db2c66944edaf8e695c3fb85cad\
  Complete output (16 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build\lib.win-amd64-3.7
  copying sha3.py -> build\lib.win-amd64-3.7
  running build_ext
  building '_pysha3' extension
  creating build\temp.win-amd64-3.7
  creating build\temp.win-amd64-3.7\Release
  creating build\temp.win-amd64-3.7\Release\Modules
  creating build\temp.win-amd64-3.7\Release\Modules\_sha3
  i:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DPY_WITH_KECCAK=1 "-IC:\Program Files\Spyder\Python\include" "-IC:\Program Files\Spyder\Python\include" "-Ii:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\ATLMFC\include" "-Ii:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\include" "-II:\Windows Kits\10\include\10.0.19041.0\ucrt" "-II:\Windows Kits\10\include\10.0.19041.0\shared" "-II:\Windows Kits\10\include\10.0.19041.0\um" "-II:\Windows Kits\10\include\10.0.19041.0\winrt" "-II:\Windows Kits\10\include\10.0.19041.0\cppwinrt" /TcModules/_sha3/sha3module.c /Fobuild\temp.win-amd64-3.7\Release\Modules/_sha3/sha3module.obj
  sha3module.c
  Modules/_sha3/sha3module.c(18): fatal error C1083: Cannot open include file: 'Python.h': No such file or directory
  error: command 'i:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Enterprise\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
  ----------------------------------------
  ERROR: Failed building wheel for pysha3
    ERROR: Command errored out with exit status 1:
     command: 'C:\Program Files\Spyder\Python\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Ali-772\\AppData\\Local\\Temp\\pip-install-1kuk9w3o\\pysha3_ddd48db2c66944edaf8e695c3fb85cad\\setup.py'"'"'; __file__='"'"'C:\\Users\\Ali-772\\AppData\\Local\\Temp\\pip-install-1kuk9w3o\\pysha3_ddd48db2c66944edaf8e695c3fb85cad\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\Ali-772\AppData\Local\Temp\pip-record-qg9wwe9i\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\Program Files\Spyder\Python\Include\pysha3'
         cwd: C:\Users\Ali-772\AppData\Local\Temp\pip-install-1kuk9w3o\pysha3_ddd48db2c66944edaf8e695c3fb85cad\
    Complete output (16 lines):
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.7
    copying sha3.py -> build\lib.win-amd64-3.7
    running build_ext
    building '_pysha3' extension
    creating build\temp.win-amd64-3.7
    creating build\temp.win-amd64-3.7\Release
    creating build\temp.win-amd64-3.7\Release\Modules
    creating build\temp.win-amd64-3.7\Release\Modules\_sha3
    i:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\bin\HostX86\x64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -DPY_WITH_KECCAK=1 "-IC:\Program Files\Spyder\Python\include" "-IC:\Program Files\Spyder\Python\include" "-Ii:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\ATLMFC\include" "-Ii:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.29.30133\include" "-II:\Windows Kits\10\include\10.0.19041.0\ucrt" "-II:\Windows Kits\10\include\10.0.19041.0\shared" "-II:\Windows Kits\10\include\10.0.19041.0\um" "-II:\Windows Kits\10\include\10.0.19041.0\winrt" "-II:\Windows Kits\10\include\10.0.19041.0\cppwinrt" /TcModules/_sha3/sha3module.c /Fobuild\temp.win-amd64-3.7\Release\Modules/_sha3/sha3module.obj
    sha3module.c
    Modules/_sha3/sha3module.c(18): fatal error C1083: Cannot open include file: 'Python.h': No such file or directory
    error: command 'i:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Enterprise\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX86\\x64\\cl.exe' failed with exit status 2
    ----------------------------------------
ERROR: Command errored out with exit status 1: 'C:\Program Files\Spyder\Python\python.exe' -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'C:\\Users\\Ali-772\\AppData\\Local\\Temp\\pip-install-1kuk9w3o\\pysha3_ddd48db2c66944edaf8e695c3fb85cad\\setup.py'"'"'; __file__='"'"'C:\\Users\\Ali-772\\AppData\\Local\\Temp\\pip-install-1kuk9w3o\\pysha3_ddd48db2c66944edaf8e695c3fb85cad\\setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record 'C:\Users\Ali-772\AppData\Local\Temp\pip-record-qg9wwe9i\install-record.txt' --single-version-externally-managed --compile --install-headers 'C:\Program Files\Spyder\Python\Include\pysha3' Check the logs for full command output.

**Comment by ikamran** [comment]


**Reply by meherett** [reply]
Hey @ikamran,

Currently, all the tests are passed but it needs a security audit report. I'm also working on that. So, you can use it as a development library. Once I finished the report I will announce it.

Thanks!

**Reply by meherett** [reply]
To create an HD wallet you don't need to download the blockchain's full node you can use libraries. For transaction and notifications systems to use, there are so many free API endpoints. eg: https://www.sochain.com
